src/main/java/eu/dissco/core/translator/service/DwcaService.java

```diff
@@ -149,6 +161,74 @@ private void processDigitalSpecimen(Collection<ObjectNode> fullRecords)
     }
   }
 
+  private void retrieveTitle(XMLEventReader xmlEventReader, HashMap<String, String> emlData)
+      throws XMLStreamException {
+    var title = xmlEventReader.nextEvent().asCharacters().getData();
+    if (title != null) {
+      log.debug("Found the dataset title in eml: {}", title);
+      emlData.put("eml:title", title);
+    }
+  }
+
+  private void addEmlDataToRecord(ObjectNode fullRecord, Map<String, String> emlData) {
+    for (var entry : emlData.entrySet()) {
+      fullRecord.put(entry.getKey(), entry.getValue());
+    }
+  }
+
+  private Map<String, String> addDatasetMeta(File mf) {
+    var emlData = new HashMap<String, String>();
+    if (mf.exists()) {
```

Comment:
Maybe make this return an Optional<Map<String, String>> instead of returning an empty map, could make code a bit more clear